### ISSUE TITLE
Random map generator

### ISSUE THREAD

**Issue by fralonra** [open-source-ideas/open-source-ideas#100]
## Project description

The idea comes from my personal interest: making maps. I want to make a tool that can generate random but logical maps with legends of real maps on them (If you find something like this, please tell me).

I made a [demo](https://github.com/fralonra/zatlas) a few months ago on generating random heightmaps, though it's quite crude. Heightmap is just a begining, and we can do more. But I know almost nothing of computer cartography, and I'm here to seek for help.

## Relevant Technology

* Any language.
* Computer cartography.
* Algorithmic knowlege.

**Comment by KOLANICH** [comment]
**Disclaimer: I'm NOT  a pro in ml and nn and have never trained a one**

I guess that using neural networks for that may be good (it depends on your goals (precomputation vs generation in runtime) and on the equipment you suppose your app work).
The paper you may need is  http://cg.cs.uni-bonn.de/en/publications/paper-details/hartmann-2017-StreetGAN/ (disclaimer - I have only read its abstracts).

There are lot of papers and competitions on aerial images classification.
I also remember there were some papers on synthesis of realistic aeroimages from labels though I have not managed to find it.

**Comment by FredrikAugust** [comment]
Procedural generation is quite a good fit for this. Don't remember the names of techniques used to create random maps, but should be easy to find.

**Comment by fralonra** [comment]
@KOLANICH @FredrikAugust @Kaijiro Thanks for all your advise!

**Comment by lukasbindreiter** [comment]
I stumbled upon [this](https://mewo2.com/notes/terrain/) blog post a while ago, which describes how to randomly generate a map including rivers, cities and so on and may be of interest:

Here is an example of how this looks like:
![Example map](https://mewo2.com/deserts/map.png)

**Comment by FredrikAugust** [comment]
PerlinNoise was the algorithm I was thinking about! Relatively simple to implement, and produces quite good results.

**Comment by fralonra** [comment]
@Enderdead It just my personal interest to generate a map that looks like real.

**Comment by Kreijstal** [comment]
what is wrong with https://azgaar.github.io/Fantasy-Map-Generator/
